### PR TITLE
[FIX] Fixed clicking on a group header outside of text not working

### LINKED ISSUE #10558: Can't click on 'Workloads' group header to go to Workloads
This is a bug specific to group headers in the side nav.

Go to the local cluster and click on Workloads.

Click on CronJobs.

Now try and click on Workloads - not by clicking on the text, but in the area around it (see red line below) - although the cursor shows a hand icon and although this works on the first click into Workloads, this does not work when the group is expanded - you have to click on the text.

Screenshot 2024-03-05 at 12 00 15

You should be able to click anywhere in the sider nav group header to go back to it.

### PR DESCRIPTION
**Pull request by eva-vashkevich**
### Summary
Fixes #10558

The link element had a padding around that wasn't clickable. Making the link the same size as the tab header fixed the problem.

### Occurred changes and/or fixed issues

I wasn't able to create a test that was failing before the change, so manual testing is required.
Some tests in cypress/e2e/tests/navigation/side-nav/product-side-nav.spec.ts were disabled pending #5966, which is now closed, so i enabled them back.

### Areas or cases that should be tested

1. Go Local -> Workloads
2. Click on "Jobs" subgroup
3. Click back on "Workloads" below from the text
4. Workloads page should be opened

### Areas which could experience regressions

N/A

### Screenshot/Video

https://github.com/rancher/dashboard/assets/133266076/d0154e67-9d22-45f0-8929-06cd71f67d7c

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/components/nav/Group.vue [hunk 1]
```diff
@@ -339,6 +341,7 @@ export default {
         > H6 {
           text-transform: none;
           padding-left: 16px;
+          padding: 8px 0px 8px 16px;
```
Comment (rak-phillip) [inline shell/components/nav/Group.vue:344]: It looks like `padding-left` is made redundant now.

```suggestion
          padding: 8px 0px 8px 16px;
```

File: shell/components/nav/Group.vue [hunk 2]
```diff
@@ -293,7 +293,8 @@ export default {
 
     > A {
       display: block;
-      padding-left: 16px;
+      box-sizing:border-box;
+      height: 33px;
```
Comment (rak-phillip) [inline shell/components/nav/Group.vue:297]: Would we be able to specify a height of `100%` since our header already has a height of `33px`?

**Review by rak-phillip — COMMENTED** [review]
Looks good and functions as advertised 🙂 

I just have a few questions before merging.

File: shell/components/nav/Group.vue [hunk 3]
```diff
@@ -302,6 +303,7 @@ export default {
       }
       > H6 {
         text-transform: none;
+        padding: 8px 0px 8px 16px;
```
Comment (rak-phillip) [inline shell/components/nav/Group.vue:306]: A bit of a nitpick: the uom is redundant with a 0 value

```suggestion
        padding: 8px 0 8px 16px;
```

File: shell/components/nav/Group.vue (hunk 2 above)
Comment (eva-vashkevich) [reply shell/components/nav/Group.vue:297]: That works too, you are right that it's better in case outer height changes

**Review by rak-phillip — APPROVED** [review]

**Pull request merged**